backends/qualcomm/utils/utils.py

```diff
@@ -37,12 +50,28 @@ def capture_program(
     module: torch.nn.Module,
     inputs: Tuple[torch.Tensor],
 ) -> exir.ExirExportedProgram:
-    edge_prog = exir.capture(
+    ex_prog = exir.capture(
         module,
         inputs,
         qnn_capture_config(),
     ).to_edge(qnn_edge_config())
-    return edge_prog.transform(qnn_partitioner_passes)
+
+    # currently ExirExportedProgram.transform does not accept
+    # changes of input number which was caused by FoldQDQ
+    # apply passes one by one here to avoid IR capture failure
+    edge_program = ex_prog.exported_program
+    graph_module = edge_program.graph_module
+    AddmmToLinearTransform()(graph_module)
```

Comment:
You can also use this https://github.com/pytorch/pytorch/blob/main/torch/export/exported_program.py#L480 to apply transformation passes.